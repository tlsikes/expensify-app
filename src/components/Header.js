import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <div>
            <NavLink to="/" activeClassName="current-link" exact={true}>Dashboard</NavLink> &nbsp;&nbsp;
            <NavLink to="/create" activeClassName="current-link">New Expense</NavLink> &nbsp;&nbsp;
            <NavLink to="/help" activeClassName="current-link">Help</NavLink> &nbsp;&nbsp;
            <p />
        </div>
    </header>
)

export default Header;
