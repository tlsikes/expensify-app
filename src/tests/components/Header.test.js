import React from 'react';
import { shallow } from 'enzyme';
import Header from "../../components/Header";

test('should render Header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();

    //example:
    //expect(wrapper.find('selector').text()).toBe('Sometext');

    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();

});