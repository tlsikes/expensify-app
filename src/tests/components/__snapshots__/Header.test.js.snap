// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <div>
    <NavLink
      activeClassName="current-link"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
       
    <NavLink
      activeClassName="current-link"
      ariaCurrent="true"
      to="/create"
    >
      New Expense
    </NavLink>
       
    <NavLink
      activeClassName="current-link"
      ariaCurrent="true"
      to="/help"
    >
      Help
    </NavLink>
       
    <p />
  </div>
</header>
`;
